vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md

```diff
@@ -7,117 +7,175 @@
 
+For the initial release of in-place updates with VPA, in-place updates will only be available
+using the `InPlaceOrRecreate` mode. In the future, once the SIG feels that the feature is
+mature enough, this behavior will become the default behavior for the `Auto` mode. See the [`Auto`
+mode documentation].
+
+[`Auto` mode documentation]: https://github.com/kubernetes/autoscaler/blob/78c8173b979316f892327022d53369760b000210/vertical-pod-autoscaler/docs/api.md#updatemode
+
 ## Context
 
-[In-place update of pod resources KEP] is available in alpha in [Kubernetes 1.27]. The feature allows changing container
-resources while the container is running. It also adds [`ResizePolicy`] field to Container. This field indicates for an
-individual resource if a container needs to be restarted by kubelet when the resource is changed. For example it may be
-the case that a Container automatically adapts to a change in CPU, but needs to be restarted for a change in Memory to
-take effect.
+[In-place update of pod resources KEP] is available in alpha in 1.27 and graduating to beta in
+1.33. The feature allows changing container resources while the container is running. It adds
+two key features:
+
+* A [`/resize` subresource] that can be used to mutate the `Pod.Spec.Containers[i].Resources`
+  field.
+* A [`ResizePolicy`] field to Container. This field allows to the user to specify the behavior when
+  modifying a resource value. Currently it has two modes:
+  - `PreferNoRestart` (default) which indicates to the container runtime that it should try to resize
+    the container without restarting. However, it does not guarantee that a restart will not
+    happen.
+  - `RestartContainer` which indicates that any mutation to the resource requires a restart (for
+    example, this is important for Java apps using the `-xmxN` which are unable to resize memory
+    without restarting).
+
+Note that resize operations will NOT change the pod's quality of service (QoS) class.
+
+Note that in the initial Beta version of in-place updates, [memory limit downscaling is forbidden]
+for pods with `resizePolicy: PreferNoRestart`. This means that when VPA will attempt to apply the
+patch, it will fail and VPA will need to fallback to a regular eviction (see below).
 
 [In-place update of pod resources KEP]: https://github.com/kubernetes/enhancements/issues/1287
-[Kubernetes 1.27]: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#api-change-3
-[`ResizePolicy`]: https://github.com/kubernetes/api/blob/8360d82aecbc72aa039281a394ebed2eaf0c0ccc/core/v1/types.go#L2448
+[`/resize` subresource]:https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#api-changes
+[`ResizePolicy`]: https://github.com/kubernetes/api/blob/4dccc5e86b957cea946a63c4f052ee7dec3946ce/core/v1/types.go#L2636
+[memory limit downscaling is forbidden]: https://github.com/kubernetes/enhancements/pull/5089
```

Comment:
```suggestion
[memory limit downscaling is forbidden]: https://github.com/kubernetes/enhancements/blob/4c3449517e1a7cf00907fd8a6dc780350275035a/keps/sig-node/1287-in-place-update-pod-resources/README.md#memory-limit-decreases
```

This may be easier to link to?

Reply:
Done.